### PR TITLE
Use target and pose directly from Echo request

### PR DESCRIPTION
**Pull request by aestene**
Note that the call to STID and the `InspectionTarget` in `MissionTask` has not been removed yet. It was decided to do that in another PR to avoid too many breaking changes in one.

### CODE REVIEW

File: backend/api/Controllers/Models/EchoInspection.cs [hunk 1]
```diff
@@ -43,18 +47,23 @@ public class EchoInspectionComparer : IEqualityComparer<EchoInspection>
         public bool Equals(EchoInspection? e1, EchoInspection? e2)
         {
             if (ReferenceEquals(e1, e2))
+            {
```
Comment (andchiind) [inline backend/api/Controllers/Models/EchoInspection.cs:50]: All these extra lines will slow down the program immensely 🤢

File: backend/api/Controllers/Models/EchoInspection.cs [hunk 2]
```diff
@@ -43,18 +47,23 @@ public class EchoInspectionComparer : IEqualityComparer<EchoInspection>
...
 
             if (e2 is null || e1 is null)
+            {
                 return false;
+            }
 
             return e1.InspectionType == e2.InspectionType
-                && e1.TimeInSeconds == e2.TimeInSeconds;
+                   && e1.TimeInSeconds == e2.TimeInSeconds
+                   && e1.InspectionPoint == e2.InspectionPoint;
```
Comment (andchiind) [inline backend/api/Controllers/Models/EchoInspection.cs:61]: Will this work even if the Position class does not override an Equals method?

File: backend/api/Services/Models/EchoPoseResponse.cs
Comment (andchiind) [inline backend/api/Services/Models/EchoPoseResponse.cs:1]: This file should probably be renamed

File: backend/api/Migrations/20231013121440_AddInspectionTargetToInspection.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace Api.Migrations
+{
+    /// <inheritdoc />
+    public partial class AddInspectionTargetToInspection : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<float>(
```
Comment (andchiind) [inline backend/api/Migrations/20231013121440_AddInspectionTargetToInspection.cs:13]: Will these columns be populated manually? (With an SQL script I mean)

**Review by andchiind — COMMENTED** [review]
Looks good

File: backend/api/Controllers/Models/EchoInspection.cs (hunk 2 above)
Comment (aestene) [reply backend/api/Controllers/Models/EchoInspection.cs:61]: Good catch and I'm not sure. Decided to override the equals method the same way as has been done for orientation.

File: backend/api/Controllers/Models/EchoInspection.cs (hunk 1 above)
Comment (aestene) [reply backend/api/Controllers/Models/EchoInspection.cs]: Fixed

File: backend/api/Services/Models/EchoPoseResponse.cs
Comment (aestene) [reply backend/api/Services/Models/EchoPoseResponse.cs:1]: Done

**Review by andchiind — APPROVED** [review]
Looks good to me, and seems to work well.

File: backend/api/Controllers/Models/EchoInspection.cs [hunk 4]
```diff
@@ -42,19 +46,18 @@ public class EchoInspectionComparer : IEqualityComparer<EchoInspection>
...
             return e1.InspectionType == e2.InspectionType
-                && e1.TimeInSeconds == e2.TimeInSeconds;
+                   && e1.TimeInSeconds == e2.TimeInSeconds
+                   && e1.InspectionPoint.Equals(e2.InspectionPoint);
         }
 
         public int GetHashCode(EchoInspection e)
         {
-            // We cannot incorporate TimeInSeconds here are SQL queries do not handle 
+            // We cannot incorporate TimeInSeconds here are SQL queries do not handle
```
Comment (Christdej) [inline backend/api/Controllers/Models/EchoInspection.cs:60]:
```suggestion
            // We cannot incorporate TimeInSeconds here as SQL queries do not handle
```

**Review by Christdej — APPROVED** [review]
LGTM

**Pull request merged**